### DISCUSSION TITLE
3D-Secure Auth iFrame

### DISCUSSION THREAD

**Discussion by rossparachute** [dojo-engineering/feedback#12] [Q&A]
Hey folks,

I noticed when using the Dojo Wallet component on confirming the payment via Google Pay's modal window that a 3D-Secure iFrame gets output.

Currently it's getting inserted just before the body closes.

Is there a way to configure where this iFrame element is appended?

Cheers,
Ross

![image](https://github.com/dojo-engineering/feedback/assets/67894901/7377dc6a-6547-418e-ac01-bee772d9fa12)

![image](https://github.com/dojo-engineering/feedback/assets/67894901/62e7f26b-ed0a-42e4-b08f-680e2b52ffe4)

**Comment by martinhay1** [comment]
Hi Ross,
Thanks for your query. Did you use a real card or a test card in this example? I believe Google Pay should be a frictionless 3DS2 transaction in production and not prompt for the 2FA with this modal.
Thanks,
Martin

**Comment by RomanKondratyev** [comment]
Hey @rossparachute ,

Could you check this example: https://github.com/dojo-engineering/dojo-samples/blob/main/samples/accept-a-payment/prebuilt-checkout/client/react-cra/src/iframe.threeds.css

Please let us know if that was not helpful.

**Comment by rossparachute** [comment]
Yeah, I was in test mode for that. The styles have sorted things.

Thanks to you both,
Ross

On Thu, Aug 3, 2023 at 10:18 AM Roman Kondratyev ***@***.***>
wrote:

> Hey @rossparachute <https://github.com/rossparachute> ,
>
> Could you check this example:
> https://github.com/dojo-engineering/dojo-samples/blob/main/samples/accept-a-payment/prebuilt-checkout/client/react-cra/src/iframe.threeds.css
>
> Please let us know if that was not helpful.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/orgs/dojo-engineering/discussions/12#discussioncomment-6625646>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AQF745OCNYHDZX5CKUNOVX3XTNUFXANCNFSM6AAAAAA27X27RQ>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***
> com>
>


-- 


Ross Hendry

Director of Software

t. 0141 846 1765

***@***.*** | www.parachute.net <https://www.thisisparachute.com/>

<https://www.facebook.com/ThisIsParachute/>
<https://twitter.com/ThisIsParachute>
<https://www.instagram.com/thisisparachute/?hl=en>
<https://www.linkedin.com/company/parachute-digital-solutions-limited/>
***@***.***,-4.2789445,17z/data=!3m1!4b1!4m5!3m4!1s0x48884432181689e7:0x4857727e622e8aea!8m2!3d55.8741653!4d-4.2767505>

Confidentiality Note: This email may contain confidential and/or private
information.

If you received this email in error please delete and notify sender.

**Comment by mbogdanel** [comment]
Hello there,

To style the card and wallet containers, we can use the config styles. How can we style the 3DS pop-up? The css above does a good job of positioning the iframe but I need some styles pulled into the box. 

Thanks

![Screenshot 2023-11-22 at 14 14 42](https://github.com/dojo-engineering/feedback/assets/61619920/cd091d8b-8cae-4897-b277-6516c2d59b6e)

**Reply by martinhay1** [reply]
Hi @mbogdanel 
The pop-up in your screenshot is Dojo's 3DS2 emulator in test mode. With production transactions the content of the modal is provided by the card issuer and is not able to be edited.
Hope this helps. Any questions let us know, thanks!

**Reply by mbogdanel** [reply]
Hi Martin
Thank you for your response.
I have managed to test locally a successful and a declined payment without 3DS with the test card numbers provided on the go-live checklist page. Couldn't test a successful payment with 3DS, do I need a code for the verification?

**Reply by rossparachute** [reply]
At least for us the code on the cards usually matched the test cards mentioned [here](https://docs.dojo.tech/payments/development-resources/testing).

For testing Google Pay we used the [Test Card Suite](https://developers.google.com/pay/api/android/guides/resources/test-card-suite). Prior to that my own card worked in test mode. The Google Pay modal that pops up via the browser rather than an element on the page should note it's in test mode and there will be no charge.

Initially we managed to get the OTP: 1234 working in the 3DS modal that popped up for Google Pay. I think we did a similar thing for it with the card component too.

However, since then - 1234 doesn't appear to be valid for any of the cards in the Google test suite anymore.

Happy if anyone finds out what the valid OTP is for Google Pay testing!

**Reply by martinhay1** [reply]
Hi @mbogdanel 
The code for the Dojo 3DS2 emulator is 1234
